tasks/components/layout/PatientDetails.tsx

```diff
@@ -64,14 +73,48 @@ export const PatientDetail = ({
   const [newPatient, setNewPatient] = useState<PatientDTO>(patient)
   const [newTask, setNewTask] = useState<TaskDTO | undefined>(undefined)
   const [boardObject, setBoardObject] = useState<KanbanBoardObject>({ searchValue: '' })
-  const sortedTasks = {
+  const [updateTimer, setUpdateTimer] = useState<NodeJS.Timer | undefined>(undefined)
+  const [notificationTimer, setNotificationTimer] = useState<NodeJS.Timer | undefined>(undefined)
+  const [sortedTasks, setSortedTasks] = useState<SortedTasks>({
     unscheduled: newPatient.tasks.filter(value => value.status === 'unscheduled'),
     inProgress: newPatient.tasks.filter(value => value.status === 'inProgress'),
     done: newPatient.tasks.filter(value => value.status === 'done'),
+  })
+  // Delete interval timer afterwards
+  useEffect(() => {
+    return () => {
+      clearInterval(updateTimer)
+      clearInterval(notificationTimer)
+    }
+  })
+
+  const changeWithUpdateTimer = (newPatient: PatientDTO) => {
+    setNewPatient(newPatient)
+    clearInterval(updateTimer)
+    // No patient validation, maybe check whether this is right
+    setUpdateTimer(setInterval(() => {
+      onUpdate(newPatient)
+      // Show Saved Notification for fade animation duration
+      clearInterval(notificationTimer)
+      setNotificationTimer(undefined)
+      setNotificationTimer(setInterval(() => {
+        clearInterval(notificationTimer)
+        setNotificationTimer(undefined)
+      }, 3000))
+    }, 3000))
```

Comment:
We need to find a way to generalize this. The debounce and notification logic can be separated. The debounce can be moved to a separate hook, like [useDebounce()](https://usehooks-ts.com/react-hook/use-debounce). With that the notification logic can be handled inside our `useDebounce()` callback.